lib/AST/LoopRefine.cpp

```diff
@@ -100,10 +100,12 @@ class DoWhileRule : public InferenceRule {
     auto cond{ifstmt->getCond()};
     std::vector<clang::Stmt *> new_body(comp->body_begin(),
                                         comp->body_end() - 1);
+    ASTBuilder ast(unit);
     if (auto else_stmt = ifstmt->getElse()) {
-      new_body.push_back(else_stmt);
+      auto cond_inv{ast.CreateLNot(cond)};
+      CopyProvenance(cond, cond_inv, provenance.use_provenance);
+      new_body.push_back(ast.CreateIf(cond_inv, else_stmt));
     }
-    ASTBuilder ast(unit);
     auto cond_inv{ast.CreateLNot(cond)};
```

Comment:
Am I right that we have `auto cond_inv{ast.CreateLNot(cond)}` twice here because of provenance tracking?